### PR TITLE
Add a failing test for version conflicts

### PR DESCRIPTION
**Pull request by magicmark**
**Summary**
This is a failing test to highlight the issue the issue here https://github.com/yarnpkg/yarn/issues/3780 https://github.com/mxmul/yarn_flat_regression

tl;dr:

```
a => b => left-pad@>=0.0.1
a => left-pad@1.0.0
```

yarn fails when doing `--flat` with the following:

```
info Unable to find a suitable version for "left-pad", please choose one by typing one of the numbers below:
  1) "left-pad@>=0.0.1" which resolved to "1.1.3"
  2) "left-pad@1.0.0" which resolved to "1.0.0"
```

(yarn version: 0.27.1)

(this seems to be a regression introduced in 0.27.0, this worked in 0.26.1)

**Test plan**
This is a test only change.

---

This test is currently failing because it's prompting for a resolution, with the following stacktrace
```
(node:23637) PromiseRejectionHandledWarning: Promise rejection was handled asynchronously (rejection id: 2)
 FAIL  __tests__/commands/install/integration.js (264.799s)
  ● unbound transitive dependencies should not conflict with top level dependency

    Error: Can't answer a question unless a user TTY
      
      at ConsoleReporter.select (src/reporters/console/console-reporter.js:322:29)
      at src/cli/commands/install.js:932:182
      at next (native)
      at step (node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
      at node_modules/babel-runtime/helpers/asyncToGenerator.js:35:14
      at Promise.F (node_modules/core-js/library/modules/_export.js:35:28)
      at node_modules/babel-runtime/helpers/asyncToGenerator.js:14:12
      at Install.flatten (src/cli/commands/install.js:933:2215)
      at src/cli/commands/install.js:922:3570
      at next (native)
      Console output:
       warning package.json: No license field
      warning No license field
      [1/4] Resolving packages...
      
      at __tests__/commands/_helpers.js:144:13
      at throw (native)
      at step (node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
      at node_modules/babel-runtime/helpers/asyncToGenerator.js:30:13
```

### CODE REVIEW

File: __tests__/fixtures/install/install-conflicts/b/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "version": "0.0.1",
+  "dependencies": {
+    "left-pad": ">=0.0.1"
```
Comment (bestander) [inline __tests__/fixtures/install/install-conflicts/b/package.json]: did you mean 1.0.1?

**Comment by magicmark** [comment]
Don't think so

I think >=0.0.1 and 1.0.0 should not conflict?

This test shows that they do conflict.

(looks like >=0.0.1 resolves to a hard pin for whatever the latest version is?)

This behaviour was not present prior to 0.27.0

**Comment by bestander** [comment]
I see, thanks.
This is related to this task in general then https://github.com/yarnpkg/yarn/issues/3778.
Let's make it passing for now and link that it should be reversed when https://github.com/yarnpkg/yarn/issues/3778 is resolved.

Do you want to have a go at implementing the post processing after resolution?

**Comment by magicmark** [comment]
@bestander Skipped the test, added comment linking to issue.

I'll have a go at https://github.com/yarnpkg/yarn/issues/3778 :)

**Comment by bestander** [comment]
Awesome!

**Comment by bestander** [comment]
Merged this, ping me on a fix :)

**Pull request merged**